test/cpp/ext/filters/otel/otel_plugin_test.cc

```diff
@@ -16,31 +16,227 @@
 //
 //
 
+#include "src/cpp/ext/filters/otel/otel_plugin.h"
+
 #include "api/include/opentelemetry/metrics/provider.h"
 #include "gmock/gmock.h"
 #include "gtest/gtest.h"
-#include "sdk/include/opentelemetry/sdk/metrics/meter_provider.h"
+#include "opentelemetry/sdk/metrics/meter_provider.h"
+#include "opentelemetry/sdk/metrics/metric_reader.h"
+
+#include <grpcpp/grpcpp.h>
 
 #include "test/core/util/test_config.h"
+#include "test/cpp/end2end/test_service_impl.h"
 
 namespace grpc {
 namespace testing {
 namespace {
 
-TEST(OTelPluginTest, ApiDependency) {
+std::shared_ptr<opentelemetry::sdk::metrics::MetricReader> g_reader_;
```

Comment:
It's not okay to have a `shared_ptr<>` as a global variable, as per:

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Reply:
It needed to be a data member.